app/controllers/api/v3/nodes_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Api
```

Comment:
Can we move this to a different PR? There are too many changes already in this one and it still requires more work to make it mergable, so let's not add to it anymore.